firebase-appdistribution/src/test/java/com/google/firebase/appdistribution/impl/TestUtils.java

```diff
@@ -104,24 +105,15 @@ static void awaitAsyncOperations(ExecutorService executorService) throws Interru
   /** Await a specified number of progress events on an {@link UpdateTask}. */
   static List<UpdateProgress> awaitProgressEvents(UpdateTask updateTask, int count)
```

Comment:
Aren't all callers of this prone to the problem that the task might have completed before the listener is added?
I think it might make sense to inline this everywhere (and employ a latch to prevent the race)

Reply:
Yeah I think you're right. There's only one other usage and maybe we're just getting lucky that it hasn't been failing. Updated.